extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/ThreadLocalPool.java

```diff
@@ -0,0 +1,85 @@
+package io.quarkus.reactive.datasource.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.logging.Logger;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.Pool;
+import io.vertx.sqlclient.PoolOptions;
+import io.vertx.sqlclient.PreparedQuery;
+import io.vertx.sqlclient.Query;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.SqlConnection;
+import io.vertx.sqlclient.Transaction;
+
+public abstract class ThreadLocalPool<PoolType extends Pool> implements Pool {
+
+    private static final Logger log = Logger.getLogger(ThreadLocalPool.class);
+
+    private ThreadLocal<PoolType> pool = new ThreadLocal<>();
+    private static List<Pool> threadLocalPools = new ArrayList<>();
```

Comment:
Seems like this need to be a Vector, and `final`. Or just `final` and synchronized..

Reply:
I added some synchronization, is it enough?

Reply:
Looks good. I found one more possible race, but I don't think it's possible in practice: one thread might be doing `pool()` and could get a live reference to the tlocal while another thread is closing.

This shouldn't happen while we're closing so I'd say this is good enough - I might send a follow-up PR just for fun :)

Reply:
Fair enough :)